runtime/configurationparser/ConfigurationParser.cpp

```diff
@@ -16,65 +17,89 @@ extern "C" {
-    }
-    return evaluateFunctionSymbol(tag, &arguments[0]);
-  }
+      const KORESymbol *symbol = constructor->getConstructor();
+      assert(symbol->isConcrete() && "found sort variable in initial configuration");
+      if (symbol->getName() == "\\dv") {
+        const auto sort = dynamic_cast<KORECompositeSort *>(symbol->getFormalArguments()[0].get());
+        const auto strPattern =
+          dynamic_cast<KOREStringPattern *>(constructor->getArguments()[0].get());
+        std::string contents = strPattern->getContents();
+        output.push_back(getToken(sort->getName().c_str(), contents.size(), contents.c_str()));
+        continue;
+      }
 
-  struct blockheader headerVal = getBlockHeaderForSymbol(tag);
-  size_t size = size_hdr(headerVal.hdr);
-  
-  if (size == 8) {
-    return (block *) ((uint64_t)tag << 32 | 1);
-  }
+      std::ostringstream Out;
+      symbol->print(Out);
+      uint32_t tag = getTagForSymbolName(Out.str().c_str());
+      if (isSymbolAFunction(tag) && constructor->getArguments().empty()) {
+        output.push_back(evaluateFunctionSymbol(tag, nullptr));
+        continue;
+      } else if (constructor->getArguments().empty()) {
+        output.push_back((block *) ((uint64_t)tag << 32 | 1));
+        continue;
+      }
+      construction term{tag, constructor->getArguments().size()};
+      workList.push_back(term);
+      for (const auto &child : constructor->getArguments()) {
+        workList.push_back(child.get());
+      }
+    } else {
+      uint32_t tag = std::get<construction>(current).tag;
+      size_t nchildren = std::get<construction>(current).nchildren;
+      std::vector<void *> arguments;
+      for (size_t i = 0; i < nchildren; i++) {
+        arguments.push_back(output.back());
+        output.pop_back();
+      }
+      if (isSymbolAFunction(tag)) {
+        output.push_back(evaluateFunctionSymbol(tag, &arguments[0]));
+        continue;
+      }
 
-  std::vector<void *> children;
-  for (const auto &child : constructor->getArguments()) {
-    children.push_back(allocatePatternAsConfiguration(child.get()));
-  }
+      struct blockheader headerVal = getBlockHeaderForSymbol(tag);
+      size_t size = size_hdr(headerVal.hdr);
 
-  if (symbol->getName() == "inj") {
-    uint16_t layout_code = layout_hdr(headerVal.hdr);
-    layout *data = getLayoutData(layout_code);
-    if (data->args[0].cat == SYMBOL_LAYOUT) {
-      block *child = (block *)children[0];
-      if (!((uint64_t)child & 1) && layout(child) != 0) {
-        uint32_t tag = tag_hdr(child->h.hdr);
-	if (tag >= first_inj_tag && tag <= last_inj_tag) {
-          return child;
+      if (tag >= first_inj_tag && tag <= last_inj_tag) {
+        uint16_t layout_code = layout_hdr(headerVal.hdr);
+        layout *data = getLayoutData(layout_code);
+        if (data->args[0].cat == SYMBOL_LAYOUT) {
+          block *child = (block *)arguments[0];
+          if (!((uint64_t)child & 1) && layout(child) != 0) {
```

Comment:
What does it mean that the lowest bit of `child` is 1? Is it the same as above? Why not a macro with an explanatory name?